nautobot/docs/development/core/release-checklist.md

```diff
@@ -306,3 +306,7 @@ Bumping version from 1.1.0 to 1.1.1-alpha.0
 
 !!! important
     Do not squash merge this branch into `develop`. Make sure to select `Create a merge commit` when merging in GitHub.
+
+### Redeploy demo.nautobot.com and next.demo.nautobot.com sandbox environments
+
+Afer you publish the release, you need to blow away the current demo environments and redeploy them for `demo.nautobot.com` and `next.demo.nautobot.com`. The documentation on how to build a demo environment is available in the [nautobot/sandboxes repository](https://github.com/nautobot/sandboxes/blob/main/README.md).
```

Comment:
1) Doing a new release shouldn't require redeploying `next.demo`, only the main `demo` site.
2) nautobot/sandboxes is a private repo currently, I don't know that it needs to be linked from here.